code/day2-operations/deployment/k8s/otel-agent-mtls.yaml

```diff
@@ -16,18 +16,29 @@ data:
       prometheus:
         config:
           scrape_configs:
-            # - job_name: 'db_jmx_metrics'
-            #   tls_config:
-            #     ca_file: /etc/istio-output-certs/root-cert.pem
-            #     cert_file: /etc/istio-output-certs/cert-chain.pem
-            #     insecure_skip_verify: true
-            #     key_file: /etc/istio-output-certs/key.pem
-            #   scheme: https 
-            #   scrape_interval: 15s
-            #   metrics_path: '/prometheus/metrics'
-            #   static_configs:
-            #     - targets:
-            #       - 'day2-service.day2-operations.svc.cluster.local:8091'             
+            - job_name: 'db_jmx_metrics'
+              tls_config:
+                ca_file: /etc/istio-output-certs/root-cert.pem
+                cert_file: /etc/istio-output-certs/cert-chain.pem
+                insecure_skip_verify: true
+                key_file: /etc/istio-output-certs/key.pem
+              scheme: https 
+              scrape_interval: 15s
+              metrics_path: '/prometheus/metrics'
+              kubernetes_sd_configs:
+                - role: endpoints
+                  namespaces:
+                    names:
+                      - day2-operations
+              relabel_configs:
+                - action: keep
+                  source_labels:
+                    - __meta_kubernetes_service_label_app
+                  regex: day2-service   
+                - action: keep
+                  source_labels:
+                    - __meta_kubernetes_endpoint_port_name
+                  regex: http-web                              
```

Comment:
using dynamic service discovery instead of static configuration. It would help to retrieve metrics when there are multiple pods reporting data.